ee/tabby-webserver/src/service/auth.rs

```diff
@@ -425,6 +412,41 @@ impl AuthenticationService for DbConn {
     }
 }
 
+async fn check_invitation(
+    db: &DbConn,
+    is_admin_initialized: bool,
+    invitation_code: Option<String>,
+    email: &str,
+) -> Result<Option<InvitationDAO>, RegisterError> {
+    if !is_admin_initialized {
+        return Ok(None);
+    }
+
+    let network_setting = db.read_security_setting().await?;
+    let domain = email.split_once('@').unwrap_or_default().1.to_string();
+    if network_setting
+        .allowed_register_domain_list
+        .contains(&domain)
```

Comment:
I do have concerns about this. This would allow anyone to sign up using an email on the allowlist, even if they don't have access to that inbox. If someone knows a domain on the allowlist, they could make up an email to get in. We probably want to have a system where you still need to get an invitation code if you're on the allowlist.

Reply:
We still need to ensure user actually owned that email address, that can be done by
1. through oauth validation(mail ownership has been verified by 3p)
2. add an graphql mutation as `requestInvitationEmail`, when user already have the Invitation, or user email domain is in allowlist, create the Invitation automatically.

Reply:
So should the check not be happening here, then? It sounds like we want to leave this endpoint untouched (not checking the allowlist) and just create a new one, `requestInvitationEmail` where the check happens.

Reply:
How do I do that?

Reply:
https://github.com/TabbyML/tabby/blob/2542fdeead0e924b601c45bd1bd99fe02495b380/ee/tabby-webserver/src/service/auth.rs#L360

Reply:
Will do shortly, thanks!